pip/pip-299.md

```diff
@@ -0,0 +1,81 @@
+- Delay Messages: Long-delayed and short-delayed messages are mixed, with only the short-delayed message successfully consumed and the long-delayed message not delivered. As time goes on, more and more records will be staying there.
+- Large Number of Consumers: If the number of consumers is large and each has some discrete ack records, all add up to a large number.
+- Large Number of Producers: If the number of producers is large, there might be a large data of Last Sequence ID to persist. This scenario only exists on the `pulsar.dedup` cursor.
+
+The config `managedLedgerMaxUnackedRangesToPersist`
+If the cursor metadata is too large to persist, the Broker will persist only part of the data according to the following priorities.
+- Subscription Properties. This part can't be split up; persist will fail if this part is too large to persist.
+- Last sequence ID of producers. This part can't be split up; persist will fail if this part is too large to persist.
+- Individual Deleted Message. If it is too large, only one part persists, and then the other part is maintained only in memory
+
+# Motivation
+
+Since the frequent persistence of `Individual Deleted Messages` will magnify the amount of BK Written and increase the latency of ack-response, the Broker does not immediately persist it when receiving a consumer's acknowledgment but persists it regularly. 
+
+The data of cursor metadata is recommended to be less than 5MB; if a subscription's `Individual Deleted Messages` data is too large to persist, as the program grows for a long time, there will be more and more non-persistent data. Eventually, there will be an unacceptable amount of repeated consumption of messages when the Broker restarts.
+
+# Goal
+
+## In Scope
+
+To avoid repeated consumption due to the cursor metadata being too large to persist.
+
+## Out of Scope
+
+This proposal will not care about this scenario: if so many producers make the metadata of cursor `pulsar.dedup` cannot persist, the task `Take Deduplication Snapshot` will be in vain due to the inability to persist.
+
+# High-Level Design
+
+Provide a new config named `dispatcherPauseOnAckStatePersistentEnabled`(default value is `false`) for a new feature: stop dispatch messages to clients when reaching the limitation `managedLedgerMaxUnackedRangesToPersist`.
+- If the user does not care about that Individual Deleted Messages can not be fully persistent, resulting in a large number of repeated message consumption, then it can be set to `false`.
+- If the user cares about repeated consumption, at can accept a decline in consumption speed when cursor metadata is too large to persist, it can be set to `true`.
+
+
+# Detailed Design
+### Public API
+
+**broker.conf**
+```
+/**
+ * After enabling this feature, Pulsar will stop delivery messages to clients if the cursor metadata is too large to persist, it will help to reduce the repeated consumption caused by the ack state that can not be fully persistent. Default "false".
```

Comment:
```suggestion
 * After enabling this feature, Pulsar will stop delivery messages to clients if the cursor metadata is too large to persist, it will help to reduce the duplicates caused by the ack state that can not be fully persistent. Default "false".
```